### PR TITLE
WIP: New group chats changes by @ingvar1995

### PR DESCRIPTION
**Pull request by iphydf**
This doesn't build, but I'll keep updating it, and people can make commits here to fix it.

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/847)

### CODE REVIEW

**Comment by iphydf** [comment]
@ingvar1995 how is your branch going? Can you get it to compile?

**Comment by ingvar1995** [comment]
Hi, @iphydf 
I'm very busy right now so I'm not going to continue ngc until May. As longer there are no developers willing help with this pr it will be inactive for a while.

File: toxcore/Messenger.c [hunk 1]
```diff
@@ -311,6 +322,46 @@ int32_t m_addfriend_norequest(Messenger *m, const uint8_t *real_pk)
...
+        Onion_Friend *onion_friend = &m->onion_c->friends_list[onion_friend_number];
+
+        Node_format tcp_relay[1]; // TODO: send > 1 relay?
+        int tcp_num = tcp_copy_connected_relays(chat->tcp_conn, tcp_relay, 1);
+
+        if (tcp_num == 1) {
+            memcpy(onion_friend->gc_data, tcp_relay, sizeof(Node_format));
+            memcpy(onion_friend->gc_data + sizeof(Node_format), chat->self_public_key, ENC_PUBLIC_KEY);
+            onion_friend->gc_data_length = GC_MAX_DATA_LENGTH;
+            memcpy(&chat->announced_node, tcp_relay, sizeof(Node_format));
```
Comment (ingvar1995) [inline toxcore/Messenger.c]: error: invalid conversion from ‘const void*’ to ‘void*’ [-fpermissive]

File: toxcore/Messenger.c [hunk 2]
```diff
@@ -311,6 +322,46 @@ int32_t m_addfriend_norequest(Messenger *m, const uint8_t *real_pk)
...
+            add_self_announce(m->group_announce, CHAT_ID(chat->chat_public_key), &chat->announced_node);
```
Comment (ingvar1995) [inline toxcore/Messenger.c]: error: invalid conversion from ‘const Node_format*’ to ‘Node_format*’

File: toxcore/onion_announce.c [hunk 3]
```diff
@@ -363,23 +410,136 @@ static int add_to_entries(Onion_Announce *onion_a, IP_Port ret_ip_port, const ui
...
+        nodes_length = pack_nodes(pl + 2 + ONION_PING_ID_SIZE, sizeof(nodes_list), nodes_list, num_nodes);
+
+        if (nodes_length <= 0)
+            return 1;
+    }
+
+    pl[1 + ONION_PING_ID_SIZE] = (uint8_t)num_nodes;
+
+    GC_Announces_List *gc_announces_list = onion_a->gc_announces_list;
+    GC_Peer_Announce *new_announce = add_gc_announce(gc_announces_list,
```
Comment (ingvar1995) [inline toxcore/onion_announce.c]: error: cannot convert ‘uint8_t* {aka unsigned char*}’ to ‘const Node_format*

**Pull request closed without merging**